### DISCUSSION TITLE
Roborock S4 Losing Settings

### DISCUSSION THREAD

**Discussion by agg23** [Hypfer/Valetudo#1881] [Q&A / Support]
I'm assuming this isn't necessarily a bug, hence the Discussions entry.

I recently discovered my Roborock S4 was not properly responding to the (external) automations I have set up. Upon visiting the web UI, I was greeted with the "new setup screen" where it asks if you want to persist map and some other things. My map data, zones, no go lines, and wifi connection are still there, but all of the other data (I assume this is all managed by Valetudo) has been reset to factory settings, such as MQTT.

This has now occurred several times, and while I'm not sure what the trigger is, it seems to be occurring a few days after I set it back up.

I am running the latest Valetudo (2023.08.0).

**Comment by Hypfer** [comment]
Check the log view in Valetudo for where it stores the config file. That should be a persistent location.
If it isn't then maybe your launch script is not setting the env variable for that correctly and therefore the config gets lost on each daily reboot.

If it's not daily, another reason could be that the roborock firmware thinks that it is low on storage and cleans up stuff from /mnt/data, although the config file shouldn't be affected by that as that algorithm iirc picks the largest files first.

File-system-corruption could also be an issue, maybe? Not sure. It's weird that the Valetudo binary itself is still there and so is all the rest of the robot configuration/state

**Comment by agg23** [comment]
The config is correctly stored at `/mnt/data/valetudo_config.json`. Interestingly,  `/mnt/data/valetudo_config.json.backup` was last modified 5 days before the latest config update (which took place the day I set it up, the day I created this discussion), and it contains a valid config. So something is specifically affecting `valetudo_config.json` and not the backup.

```bash
root@rockrobo:~# ls -al /mnt/data
total 197717
drwxr-xr-x 7 root root      4096 Sep 16 10:59 .
drwxr-xr-x 6 root root      1024 Jan  1  1970 ..
-rw-r--r-- 1 root root        45 Mar  9  2021 firmware.md5sum
-rwxrwxr-x 1 1001 1001      2312 Mar  9  2021 install_a.sh
-rwxrwxr-x 1 1001 1001      2326 Mar  9  2021 install_b.sh
drwx------ 2 root root     16384 Jan  1  1970 lost+found
drwxr-xr-x 2 root root      4096 Sep 22 10:58 miio
drwxr-xr-x 6 root root      4096 Sep 22 10:58 rockrobo
drwxr-xr-x 3 root root      4096 Jan  1  1970 rriot
-rw-r--r-- 1 root root 137588656 Jan  1  1970 v11_000358.tar.gz
-rw-r--r-- 1 root root  32404035 Mar 10  2021 valetudo
-rw-r--r-- 1 root root  32404043 Jan  1  1970 valetudo.master
-rw-r--r-- 1 root root      1850 Sep 20 10:59 valetudo_config.json
-rw-r--r-- 1 root root      1758 Sep 15 10:59 valetudo_config.json.backup
drwxr-xr-x 2 root root      4096 Jan  1  1970 wlan
```

Storage space isn't a problem (unless the firmware is somehow confused):

```bash
root@rockrobo:~# df -h
Filesystem       Size  Used Avail Use% Mounted on
/dev/root        494M  287M  182M  62% /
devtmpfs         202M  4.0K  202M   1% /dev
none             4.0K     0  4.0K   0% /sys/fs/cgroup
tmpfs             30M  1.7M   29M   6% /tmp
none              50M  228K   50M   1% /run
none             5.0M     0  5.0M   0% /run/lock
tmpfs            100M  2.6M   98M   3% /run/shm
none             100M     0  100M   0% /run/user
/dev/mmcblk0p11   16M  1.9M   13M  13% /mnt/reserve
/dev/mmcblk0p1   1.6G  719M  751M  49% /mnt/data
/dev/mmcblk0p10  520M   17M  477M   4% /mnt/updbuf
/dev/mmcblk0p6    62M  5.4M   54M  10% /mnt/default
```

Does the robot run `fsck` on reboot? Should I figure out how to get `/mnt/data` unmounted and run it manually?
